test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -394,36 +566,63 @@ TEST_P(ExtProcIntegrationTest, GetAndRespondImmediately) {
-  waitForFirstMessage(request_headers_msg);
-  EXPECT_TRUE(request_headers_msg.has_request_headers());
-  processor_stream_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, false);
-
-  // Response to request_headers
-  ProcessingResponse resp1;
-  resp1.mutable_request_headers();
-  processor_stream_->sendGrpcMessage(resp1);
-
+  auto response = sendDownstreamRequest(absl::nullopt);
+  processRequestHeadersMessage(true, absl::nullopt);
   handleUpstreamRequest();
 
-  // response_headers message to processor
-  ProcessingRequest response_headers_msg;
-  ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, response_headers_msg));
-  ASSERT_TRUE(response_headers_msg.has_response_headers());
+  processAndRespondImmediately(false, [](ImmediateResponse& immediate) {
+    immediate.mutable_status()->set_code(envoy::type::v3::StatusCode::Unauthorized);
+    immediate.set_body("{\"reason\": \"Not authorized\"}");
+    immediate.set_details("Failed because you are not authorized");
+  });
 
-  // Response to response_headers
-  ProcessingResponse resp2;
-  auto* immediate_response = resp2.mutable_immediate_response();
-  immediate_response->mutable_status()->set_code(envoy::type::v3::StatusCode::Unauthorized);
-  immediate_response->set_body("{\"reason\": \"Not authorized\"}");
-  immediate_response->set_details("Failed because you are not authorized");
-  processor_stream_->sendGrpcMessage(resp2);
+  verifyDownstreamResponse(*response, 401);
+  EXPECT_EQ("{\"reason\": \"Not authorized\"}", response->body());
+}
+
+// Test the filter with request body streaming enabled using
+// an ext_proc server that responds to the request_body message
+// by sending back an immediate_response message
+TEST_P(ExtProcIntegrationTest, GetAndRespondImmediatelyOnRequestBody) {
+  proto_config_.mutable_processing_mode()->set_request_body_mode(ProcessingMode::BUFFERED);
+  initializeConfig();
+  HttpIntegrationTest::initialize();
+  auto response = sendDownstreamRequestWithBody("Replace this!", absl::nullopt);
+  processRequestHeadersMessage(true, absl::nullopt);
+  processAndRespondImmediately(false, [](ImmediateResponse& immediate) {
+    immediate.mutable_status()->set_code(envoy::type::v3::StatusCode::Unauthorized);
+    immediate.set_body("{\"reason\": \"Not authorized\"}");
+    immediate.set_details("Failed because you are not authorized");
+  });
 
   verifyDownstreamResponse(*response, 401);
   EXPECT_EQ("{\"reason\": \"Not authorized\"}", response->body());
 }
 
+// Test the filter with body streaming enabled using
+// an ext_proc server that responds to the response_body message
+// by sending back an immediate_response message. Since this
+// happens after the response headers have been sent, as a result
+// Envoy should just reset the stream.
+TEST_P(ExtProcIntegrationTest, GetAndRespondImmediatelyOnResponseBody) {
+  Logger::Registry::getLog(Logger::Id::filter).set_level(spdlog::level::trace);
```

Comment:
Seems like we set this in initializeConfig already? 

Also generally I think we avoid bumping the log level in test by default, instead relying on `--test_arg='-l trace'` or whatnot to bump the level, though if you find it useful to have it enabled by default I don't feel that strongly about it

Reply:
Yes -- I'll take out the trace statement and use the "test_arg" command next time and see how it goes.